### PR TITLE
usbdev: Add low level initialization to `periph_init()`

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds low level initialization functions for `usbdev`. A low level init function is added to be called in `periph_init()`. This should prepare any configured usbdev device available on the mcu for the real initialization call. Most of this is going to be hooking up the right driver struct with the right context.

The other function is added to allow for retrieving a usbdev context. this is similar to the already existing `SPI_DEV()` and `I2C_DEV()` macros.

Both of these functions must be implemented by the cpu-specific usbdev periph code.

### Testing procedure

Must not impact anything normally as there is no board providing `periph_usbdev` yet. With `periph_usbdev` this should cause a linker error for a missing implementation of `usbdev_init_lowlevel()`

### Issues/PRs references

image
fixes a chicken/egg issue with the current usb PR set.

### CODE REVIEW

File: drivers/include/periph/usbdev.h [hunk 1]
```diff
@@ -355,6 +355,22 @@ typedef struct usbdev_driver {
...
+void usbdev_init_lowlevel(void);
+
+/**
+ * @brief Retrieve usbdev context from the peripheral
+ *
+ * @param num   usbdev peripheral number to retrieve
+ *
+ * @returns     the usbdev context at index @p num
+ */
+usbdev_t *usbdev_get_ctx(unsigned num);
```
Comment (dylad) [inline drivers/include/periph/usbdev.h]: uint8_t num ?

Comment (bergzand) [reply drivers/include/periph/usbdev.h]: I'd rather keep this `unsigned`( or `unsigned int`). The size here doesn't really matter, I'd be suprised to see an mcu with more than 255 USB peripherals. With this in mind I'd like to leave it up to the platform to pick a decent size.

**Comment by dylad** [comment]
@bergzand Are you done here ? I'd like to merge it before feature freezes.

**Comment by bergzand** [comment]
> @bergzand Are you done here ? I'd like to merge it before feature freezes.

@dylad Sorry, wasn't aware that you were waiting for me here. Unless you consider your change request above unresolved I don't have anything to add here. What do you think?

File: drivers/include/periph/usbdev.h (hunk 1 above)
Comment (haukepetersen) [reply drivers/include/periph/usbdev.h]: +1

**Comment by dylad** [comment]
Here we go !

**Comment by bergzand** [comment]
@dylad Thanks for testing!

**Pull request merged**